src/core/cluster.rs

```diff
@@ -563,13 +684,16 @@ impl<'a, T: Number, U: Number, D: Dataset<T, U>> Cluster<'a, T, U, D> {
     /// including this `Cluster`.
     pub fn subtree(&self) -> Vec<&Self> {
```

Comment:
Add a line in the doc-string saying that this is a potentially expensive function.